metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -129,155 +146,397 @@ func getResource(resourceName string) kubernetes.Resource {
 	}
 }
 
-// NewResourceMetadataEnricher returns an Enricher configured for kubernetes resource events
+// getExtraWatchers returns a list of the extra resources to watch based on some resource.
+// The full list can be seen in https://github.com/elastic/beats/issues/37243, at Expected Watchers section.
+func getExtraWatchers(resourceName string, config *kubernetesConfig) []string {
```

Comment:
Also as a general comment all this functions I think can move to util folder and keep same structure (as we have util.NewResourceMetadataEnricher), because now we go back and forth